src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/use_row_header_components.tsx

```diff
@@ -0,0 +1,523 @@
+import { type AggregateQuery } from '@kbn/es-query';
+import { appendFilteringWhereClauseForCascadeLayout, constructCascadeQuery } from '@kbn/esql-utils';
+import { css } from '@emotion/react';
+import {
+  EuiBadge,
+  EuiContextMenu,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiTextTruncate,
+  EuiWrappingPopover,
+  copyToClipboard,
+} from '@elastic/eui';
+import { NumberBadge, type DataCascadeRowProps } from '@kbn/shared-ux-document-data-cascade';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import type { UnifiedDataTableProps } from '@kbn/unified-data-table';
+import {
+  type ESQLStatsQueryMeta,
+  type SupportedStatsFunction,
+  getStatsGroupFieldType,
+  getFieldParamDefinition,
+} from '@kbn/esql-utils/src/utils/cascaded_documents_helpers';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { ESQLControlVariable } from '@kbn/esql-types';
+import type { StatsCommandSummary } from '@kbn/esql-ast/src/mutate/commands/stats';
+import { getPatternCellRenderer } from '../../../../../../context_awareness/profile_providers/common/patterns_data_source_profile/pattern_cell_renderer';
+
+import type { ESQLDataGroupNode, DataTableRecord } from './types';
+import {
+  type TabStateGlobalState,
+  internalStateActions,
+  useInternalStateDispatch,
+} from '../../../../state_management/redux';
+import type { DiscoverStateContainer } from '../../../../state_management/discover_state';
+
+interface RowContext {
+  groupId: string;
+  groupValue: string;
+}
+
+interface RowClickActionContext {
+  dataView: DataView;
+  editorQuery: AggregateQuery;
+  statsFieldSummary: StatsCommandSummary['grouping'] | undefined;
+  esqlVariables: ESQLControlVariable[] | undefined;
+  rowContext: RowContext;
+  services: UnifiedDataTableProps['services'];
+  closeActionMenu: () => void;
+  globalState: TabStateGlobalState;
+  openInNewTab: (...args: Parameters<typeof internalStateActions.openInNewTab>) => void;
+  updateESQLQuery: DiscoverStateContainer['actions']['updateESQLQuery'];
+}
+
+/**
+ * Defines the context menu actions for the row header.
+ */
+const contextRowActions: Array<
+  EuiContextMenuPanelItemDescriptorEntry & {
+    /**
+     * The stats function type to render the action for. If not provided, the action will be rendered for all stats function types.
+     */
+    renderFor?: SupportedStatsFunction;
+    /**
+     * Allows an action to be enabled when the field is filterable.
+     */
+    enabledWhenFilterable?: boolean;
+  }
+> = [
+  {
+    name: i18n.translate('discover.esql_data_cascade.row.action.copy_to_clipboard', {
+      defaultMessage: 'Copy to clipboard',
+    }),
+    icon: 'copy',
+    'data-test-subj': 'dscCascadeRowContextActionCopyToClipboard',
+    onClick(this: RowClickActionContext) {
+      copyToClipboard(this.rowContext.groupValue as string);
```

Comment:
```suggestion
      copyToClipboard(this.rowContext.groupValue);
```
Looks like this is always a string.